src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/TypeTable.cs

```diff
@@ -23,8 +23,8 @@ public KeyPair(Key1 pKey1, Key2 pKey2)
 
         public bool Equals(KeyPair<Key1, Key2> other)
         {
-            return Equals(_pKey1, other._pKey1)
-                && Equals(_pKey2, other._pKey2);
+            return EqualityComparer<Key1>.Default.Equals(_pKey1, other._pKey1)
+                && EqualityComparer<Key2>.Default.Equals(_pKey2, other._pKey2);
```

Comment:
This is necessary if we have a `KeyPair`s typed on a `KeyPair` so as to call into this method rather than the sub-optimal one below it. Since there have been improvements in the jitting of `EqualityComparer<>.Default` it should have no downside.